### PR TITLE
Gradle tools integration (CheckStyle, FindBugs, PMD, JUnit, Cobertura) + Travic CI

### PR DESCRIPTION
**Pull request by iiKoe**
This integrates the specified tools.

-  CheckStyle, check for the code formatting. A modified Google Java rule-set is used. In the future rules can be added or changed (after the initial game is submitted). I changed the code so all rules are followed.

- FindBugs, check for possible Bugs. It now only stops a build when a ERROR condition is detected, warnings are currently ignored. I changed the code to remove errors, but some warnings still remain. They have to be looked at at another time.

- PMD, Also looks for code flaws, all warnings have been changed in the code.

- JUnit and Cobertura, these are testing tools. They are added to Gradle now, but are both not setup. This should be done after the initial game is submitted.

- Travis CI, This is a build tool that builds all the commits and puts a pass or fail mark behind it.
(some messing around was done in the last couple of commits to get unix endlines and make the gradlew script executable)

### CODE REVIEW

File: core/src/com/group66/game/BustaMove.java [hunk 1]
```diff
@@ -41,16 +41,8 @@ public int getGameWidth() {
...
-	public void create () {
+	public void create() {
 		batch = new SpriteBatch();
 		this.setScreen(new MainMenuScreen(this));
 	}
-
-	/* (non-Javadoc)
-	 * @see com.badlogic.gdx.Game#render()
-	 */
-	@Override
```
Comment (11maxed11) [inline core/src/com/group66/game/BustaMove.java]: What does this do and what are the implications of removing it?
I'm having a a difficult error in displaying text on screen, would this be somehow related?

File: core/src/com/group66/game/cannon/BallManager.java [hunk 2]
```diff
@@ -127,12 +141,14 @@ public void draw(SpriteBatch batch, float runtime) {
 				}
 			}
 			/* Does the ball hit the edge? */
-			if (ball.getX() - ball.getRadius() <= Config.BOUNCE_X_MIN && Math.toDegrees(ball.getAngle()) > 90) {
+			if (ball.getX() - ball.getRadius() <= Config.BOUNCE_X_MIN
```
Comment (11maxed11) [inline core/src/com/group66/game/cannon/BallManager.java]: A general question: I noticed that when the line is long, you write it in two lines. Many changes like this were done in this pull request. Is the the effect of some tool you integrated?

File: core/src/com/group66/game/BustaMove.java (hunk 1 above)
Comment (iiKoe) [inline core/src/com/group66/game/BustaMove.java]: This overwrites the create() function and the render function
in the render function only the super.render() was passed, so I threw it out

File: core/src/com/group66/game/helpers/AssetLoader.java [hunk 3]
```diff
@@ -12,75 +12,77 @@
...
-    /**
-     * Dispose of the textures.
-     */
-    public static void dispose() { // TODO dispose all textures
-        // We must dispose of the texture when we are finished.
-        bgTexture.dispose();
-        ballTexture.dispose();
-    }
+	/** The background texture. */
+	public static Texture bgTexture;
```
Comment (11maxed11) [inline core/src/com/group66/game/helpers/AssetLoader.java]: Was this regenerated using some tool?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+language: java
```
Comment (11maxed11) [inline .travis.yml]: You added many tools in this pull request. Would us not having these tools installed cause any build errors for us?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+jdk:
+    - oraclejdk7
```
Comment (11maxed11) [inline .travis.yml]: I believe we specified JDK 6 or higher in the non-functional requirements. please change.

Comment (iiKoe) [inline .travis.yml]: O yea, that was right, my mistake

File: .travis.yml (hunk 4 above)
Comment (iiKoe) [inline .travis.yml]: No, these tools only check the code and code style

File: .travis.yml (hunk 5 above)
Comment (iiKoe) [inline .travis.yml]: Looks like Travis-CI only supports openjdk6 for java 6 (java 6 is EOL)
This commit log is going to be a mess

File: core/src/com/group66/game/cannon/BallManager.java (hunk 2 above)
Comment (iiKoe) [inline core/src/com/group66/game/cannon/BallManager.java]: Yes, the CheckStyle tool. I implemented a modified Google Java style, as I thought that they should probably have a nice style. With that style comes a line length restraint. It was 100 originally but I put it to 120 as I found it a better number for Java. In the future we could all sit down and discuss some of the style constrains (after the initial delivery)

Comment (11maxed11) [inline core/src/com/group66/game/cannon/BallManager.java]: Nice. Does it changes the code and make it follow the style automatically too? or only points to where changes should be done?

File: .travis.yml (hunk 5 above)
Comment (11maxed11) [inline .travis.yml]: So that's why the Travis-ci builds failed?
And what to do now?

**Pull request merged**